File: c/uocsonguyentolonnhat.cpp
#include <stdio.h>
#include <math.h>

int check_snt(long long n){
    if(n < 2) return 0;
    for(long long i = 2; i <= sqrt(n); i++){
        if(n % i == 0) return 0;
    }
    return 1;
}

int main(){
    int t;
    scanf("%d", &t);
    while(t--){
        long long n;
        scanf("%lld", &n);
        long long max = 1;
        for(long long i = 2; i <= sqrt(n); i++){
            if(n % i == 0){
                while(n % i == 0){
                    n /= i;
                }
                if(check_snt(i) && i > max) max = i;
            }
        }
        if(n > 1 && check_snt(n) && n > max) max = n;
        printf("%lld\n", max);
    }
    return 0;
}

File: java/javaPogg/src/uocSoNguyenToLonNhat.java
import java.util.Scanner;

public class uocSoNguyenToLonNhat {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        while(t-- > 0){
            int n = sc.nextInt();
            int ans = -1;
            for(int i = 2; i <= Math.sqrt(n); i++){
                if(n % i == 0){
                    ans = Math.max(ans, i);
                    while(n % i == 0) n /= i;
                }
            }
            ans = Math.max(ans, n);
            System.out.println(ans);
        }
    }
}
